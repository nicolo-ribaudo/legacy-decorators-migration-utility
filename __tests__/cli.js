"use strict";

const shelljs = require("shelljs");
const path = require("path");

describe("cli", () => {
  it("works", () => {
    shelljs.cd(__dirname);
    shelljs.cd("cli-fixtures/works");
    const out = shelljs.exec(
      "node ../../../packages/wrap-legacy-decorators/src/index.js index.js --decorators-before-export --external-helpers"
    );

    expect(out.toString()).toMatchSnapshot();
  });

  it("requires --write for multiple files", () => {
    shelljs.cd(__dirname);
    shelljs.cd("cli-fixtures/multiple-files");
    const out = shelljs.exec(
      "node ../../../packages/wrap-legacy-decorators/src/index.js *.js --decorators-before-export --external-helpers"
    );

    const err = out.stderr
      .replace(path.join(__dirname, ".."), "[ROOT]")
      .replace(/\(node:\d+\)/g, "");
    expect(err).toMatchSnapshot();
  });
});
