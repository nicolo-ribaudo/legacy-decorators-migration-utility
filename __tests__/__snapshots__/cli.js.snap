// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cli requires --write for multiple files 1`] = `
" UnhandledPromiseRejectionWarning: Error: Multiple input files detected: writing the output to the stdout is not supported. You must pass the '--write' flag.
    at Command.run ([ROOT]/packages/wrap-legacy-decorators/src/index.js:72:11)
 UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
 [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
"
`;

exports[`cli works 1`] = `
"import _ from \\"decorators-compat\\";
@_([console
  .log
  .bind(console)])
export default class Foo {
  @_([function(desc){desc.initializer=()=>3}])

  baz = 2;

  @_([enumerable])
  method(

  ) {
      return  \\"strange\\" +
                \\"indentation\\";
  }
}
"
`;
